### PR TITLE
Add logcdf to distributions 

### PR DESCRIPTION
**Pull request by mborland**
From #303 and complements #762.

### CODE REVIEW

File: include/boost/math/distributions/logistic.hpp [hunk 1]
```diff
@@ -146,6 +146,49 @@ namespace boost { namespace math {
...
+       if(power > tools::log_max_value<RealType>())
+       {
+          return 0;
+       }
+       if(power < -tools::log_max_value<RealType>())
+       {
+          return 1;
+       }
+
+       return -log(1 + exp(power));
```
Comment (jzmaddock) [inline include/boost/math/distributions/logistic.hpp:191]: That should probably be `-log1p(exp(power))` to preserve a small valued result when the cdf is ~ 1

File: include/boost/math/distributions/laplace.hpp [hunk 2]
```diff
@@ -226,6 +226,50 @@ inline RealType cdf(const laplace_distribution<RealType, Policy>& dist, const Re
...
+   RealType scale( dist.scale() );
+   RealType location( dist.location() );
+
+   if (x < location)
+   {
+      result = ((x - location) / scale) - boost::math::constants::ln_two<RealType>();
+   }
+   else
+   {
+      result = log(2 - exp((location - x) / scale)) - boost::math::constants::ln_two<RealType>();
```
Comment (jzmaddock) [inline include/boost/math/distributions/laplace.hpp:268]: `log1p(-exp((location - x) / scale) / 2)` should preserve small valued results better.

**Comment by jzmaddock** [comment]
Just checking through to see which others could be usefully specialized, 

exponential distribution would be `log1p(- exp(-x * λ) )`
Geometric: `log1p(-pow(1-p, floor(x)))`
Logistic: `-log1p(exp(-(x-u)/s))`
Pareto: `log1p(-pow(beta/x, alpha))`
Rayleigh: `log1p(-exp(-x^2/2)*sigma^2)`
Weibull: `log1p(-exp(pow(-x/beta, alpha)))`

I've just typed those straight in, they need careful checking! :)

**Comment by mborland** [comment]
@jzmaddock Now that this is green could you take another look through?

File: include/boost/math/distributions/extreme_value.hpp [hunk 3]
```diff
@@ -174,6 +174,32 @@ inline RealType cdf(const extreme_value_distribution<RealType, Policy>& dist, co
...
+   if(0 == detail::verify_scale_b(function, b, &result, Policy()))
+      return result;
+   if(0 == detail::check_finite(function, a, &result, Policy()))
+      return result;
+   if(0 == detail::check_finite(function, a, &result, Policy()))
+      return result;
+   if(0 == detail::check_x("boost::math::logcdf(const extreme_value_distribution<%1%>&, %1%)", x, &result, Policy()))
+      return result;
+
+   result = -exp((a-x)/b);
```
Comment (jzmaddock) [inline include/boost/math/distributions/extreme_value.hpp:198]: Not sure if it matters, but if we're doing complements as well, then this one would be `log1p(-exp(-exp((a-x)/b)))`

**Comment by jzmaddock** [comment]
Many thanks Matt - this all looks good to go to me - feel free to merge away.

**Comment by mborland** [comment]
@jzmaddock I added the complement logcdf to the extreme value distribution and the laplace distribution since I missed the latter as well. If you'd like to review those two I can merge once this finishes.

**Pull request merged**